sound/soc/sof/intel/hda-dsp.c

```diff
@@ -868,8 +868,9 @@ int hda_dsp_suspend(struct snd_sof_dev *sdev, u32 target_state)
 		}
 
 		/* enable L1SEN to make sure the system can enter S0Ix */
-		snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, HDA_VS_INTEL_EM2,
-					HDA_VS_INTEL_EM2_L1SEN, HDA_VS_INTEL_EM2_L1SEN);
+		if (hda->l1_disabled)
```

Comment:
the commit message and this test seems to contradict each other?

Reply:
It looks correct to me. If L1 is already enabled, no need to enable it right? Anyway, this is a fixup

Reply:
I am getting a migraine with the logical inversions and not knowing if l1_disabled is a configuration or a mirrored status. It seems to be both...

Reply:
l1_disabled is set only when it DMI L1 is disabled with open capture streams. If there're no capture streams it woulne be set and therefor DMI L1 will already be enabled right?